### ISSUE TITLE
Stuff sometimes keeps rendering after cause is gone

### ISSUE THREAD

**Issue by LordEnder-Kitty** [Goby56/wakes#121]
**Describe the bug**
Certain effects sometimes persist after the water and entity in said water are no longer present in that location.

**Expected behavior**
Effects should disappear when there's no cause.

**To Reproduce**
Not really sure. It just happens occasionally while I'm playing on me and my friends server.

**Screenshots**
![image](https://github.com/user-attachments/assets/788d7ad2-02b2-43a5-bce6-ef0e1fdcdc22)

**Hardware and OS**
Linux Mint 21.3 Cinnamon 6.0.4
Old Intel CPU and GPU

**Versions**
- Minecraft 1.21.1
- Fabric 0.16.9
- Wakes v0.2.2-1.21
- Fabric API 0.110.0+1.21.1

**Comment by LordEnder-Kitty** [comment]
It just happened again!
![image](https://github.com/user-attachments/assets/699054e6-5985-43a5-853b-70a897bcbee0)

**Comment by Goby56** [comment]
Update to a newer version of Wakes

**Comment by LordEnder-Kitty** [comment]
> Update to a newer version of Wakes

Waiting for the update to 1.21.4

**Comment by Goby56** [comment]
Unfortunately I can't update the mod to that version as Satin hasn't been updated yet. 1.21.2 made some pretty large changes to the rendering pipeline and it will take a while for Satin to catch up. Sorry for the inconvenience.

**Comment by LordEnder-Kitty** [comment]
Yeah. I looked at your 1.21.4 branch and saw the commit name and went to check the Satin github and they're working on it.
